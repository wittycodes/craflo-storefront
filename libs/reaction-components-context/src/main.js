import React from "react";
import hoistNonReactStatics from "hoist-non-react-statics";

/**
 * @see https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging
 * @param {Object} HOC The higher-order component
 * @param {Object} WrappedComponent The wrapped component
 * @returns {undefined}
 */
function wrapDisplayName(HOC, WrappedComponent) {
  const innerDisplayName = WrappedComponent.displayName || WrappedComponent.name || "Component";
  HOC.displayName = `WithComponents(${innerDisplayName})`;
}

const ComponentsContext = React.createContext();

export const ComponentsProvider = ComponentsContext.Provider;

/**
 * @name withComponents
 * @method
 * @param {React.Component} Component A React component class
 * @param {String} [prefix] A prefix to look for on keys of the components object, to
 *   override the primary component. Default is `Component.name`
 * @returns {React.Component} A new React component that wraps the passed in component,
 *   passing it a `components` prop from the nearest `ComponentsProvider`. If the
 *   `components` prop is provided already, the two objects are merged with
 *   the prop overriding the context.
 */
export function withComponents(Component, prefix = Component.name) {
  const WithComponents = React.forwardRef((props, ref) => (
    <ComponentsContext.Consumer>
      {(componentsFromContext) => {
        const { components: componentsFromProps, ...otherProps } = props;

        let components = null;
        if (componentsFromContext || componentsFromProps) {
          components = {
            ...(componentsFromContext || {}),
            ...(componentsFromProps || {})
          };

          // Look for prefix overrides. For example, wrapping a component named
          // ContactForm that uses the `Button` component, it would look for
          // `ContactForm_Button` component and pass that as the `components.Button` instead
          // if found.
          if (prefix) {
            Object.keys(componentsFromContext || {}).forEach((name) => {
              const namePieces = name.split("_");
              if (namePieces[0] === prefix) {
                components[namePieces[1]] = componentsFromContext[name];
              }
            });
          }
        }

        return <Component ref={ref} {...otherProps} components={components} />;
      }}
    </ComponentsContext.Consumer>
  ));

  // https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over
  hoistNonReactStatics(WithComponents, Component);

  // https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging
  wrapDisplayName(WithComponents, Component);

  return WithComponents;
}
