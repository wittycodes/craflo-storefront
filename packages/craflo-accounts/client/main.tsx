import React, { createContext, Suspense } from "react";
import ReactDOM from "react-dom";
import { ApolloProvider } from "react-apollo";
import { ApolloClient } from "apollo-client";
import { HttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { BrowserRouter, Route } from "react-router-dom";
import { Meteor } from "meteor/meteor";
import App from "./components/App";
import Loading from "./components/Loading";
import Logger from "./Logger";

const RouterContext = createContext(null);

/**
 * @summary Asks the server what the GraphQL API URL is and returns it.
 * @return {Promise<String>} The GraphQL API URL
 */
function getGraphQlApiUrl() {
  return new Promise((resolve, reject) => {
    Meteor.call("getGraphQLApiUrl", (error, result) => {
      if (error) {
        reject(error);
      } else {
        resolve(result);
      }
    });
  });
}

/**
 * @summary Initialize the app
 * @return {Promise<undefined>} Nothing
 */
async function init() {
  const graphQlApiUrl = await getGraphQlApiUrl();

  const apolloClient = new ApolloClient({
    link: new HttpLink({ uri: graphQlApiUrl }),
    cache: new InMemoryCache()
  });

  ReactDOM.render(
    (
      <Suspense fallback={<Loading />}>
        <ApolloProvider client={apolloClient}>
          <BrowserRouter>
            <Route>
              {(routeProps) => (
                <RouterContext.Provider value={routeProps}>
                  <App />
                </RouterContext.Provider>
              )}
            </Route>
          </BrowserRouter>
        </ApolloProvider>
      </Suspense>
    ),
    document.getElementById("react-root")
  );
}

init().catch((error) => {
  Logger.error(error);
});
