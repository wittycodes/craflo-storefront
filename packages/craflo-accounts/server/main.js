import { Meteor } from "meteor/meteor";
import {ServiceConfiguration} from 'meteor/service-configuration'
import Logger from "@reactioncommerce/logger";
import config from "./config.js";
import { oauthLogin } from "./oauthMethods.js";

Meteor.methods({
  "getGraphQLApiUrl": () => config.API_URL,
  "oauth/login": oauthLogin
});

Meteor.methods({
  // to
  "sendOtpForLogin": function(toPhone) {
    if (Meteor.isClient) return null;

    // otp must be generated on the server and never revealed to the client
    check(toPhone, String);

    // send otp as sms
    let otp = Math.round(Random.fraction() * 100000);
    console.log(otp)
    // Accounts.setPhoneOtp sets the otp in the `__otps` collection: {phone, otp, purpose: '__login__'}.
    Accounts.setPhoneOtp(toPhone, otp);
  },
});


// Init endpoints
import "./i18n/handler.js";
import "./oauthEndpoints.js";

Meteor.startup(() => {
  ServiceConfiguration.configurations.upsert({
    service: "google"
  }, {
    $set: {
      clientId: "683466154352-c7lbivd6l39m79kmpgm6pb7ju1pvmo4e.apps.googleusercontent.com",
      loginStyle: "popup",
      secret: "xCPKyq_fCfUuCJwHsf7pnpzh"
    }
  });
  Logger.info(`Serving Reaction Identity at ${config.ROOT_URL}`);
});
