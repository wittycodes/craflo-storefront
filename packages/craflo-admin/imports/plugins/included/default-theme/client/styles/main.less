/**
 * WARNING - DEPRECATED
 * These styles are a leftover from the Meteor-only all-in-one UI. Some of them are still needed for global styles and
 * components that haven't been ported over to Material UI yet.
 */


// -----------------------------------------------------------------------------
// Import Bootstrap 3 Styles
//

// Core variables and mixins
@import "{}/node_modules/bootstrap/less/variables.less";
@import "{}/node_modules/bootstrap/less/mixins.less";

// Reset and dependencies
@import "{}/node_modules/bootstrap/less/normalize.less";

// Core CSS
@import "{}/node_modules/bootstrap/less/scaffolding.less";
@import "{}/node_modules/bootstrap/less/type.less";
@import "{}/node_modules/bootstrap/less/tables.less";
@import "{}/node_modules/bootstrap/less/buttons.less";

// Components
@import "{}/node_modules/bootstrap/less/panels.less";

// -----------------------------------------------------------------------------
// Reaction Core theme files
//

// Base Styles
@import "base.less";
@import "button.less";
@import "cardGroup.less";
@import "cards.less";
@import "fonts.less";
@import "mixins.less";
@import "panels.less";
@import "sortableTable.less";
@import "textfield.less";
@import "variables.less";

// Dashboard
@import "dashboard/emailConfig.less";
@import "dashboard/shippingConfig.less";
@import "dashboard/tables.less";
