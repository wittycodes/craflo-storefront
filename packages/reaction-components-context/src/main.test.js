import React, { Component } from "react";
import renderer from "react-test-renderer";
import { ComponentsProvider, withComponents } from "./main";

class MyComponent extends Component {
  static someStaticProp = "foo";
  static displayName = "MyComponent";

  render() {
    return JSON.stringify(this.props.components);
  }
}

const MyWrappedComponent = withComponents(MyComponent);

test("passes components from context in components prop", () => {
  const contextComponents = { foo: "bar" };
  const component = renderer.create(<ComponentsProvider value={contextComponents}><MyWrappedComponent /></ComponentsProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("merges components from context with components prop", () => {
  const contextComponents = { abc: "123" };
  const instanceComponents = { xyz: "321" };
  const component = renderer.create(<ComponentsProvider value={contextComponents}><MyWrappedComponent components={instanceComponents} /></ComponentsProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("components prop key overwrites context key", () => {
  const contextComponents = { Button: "div" };
  const instanceComponents = { Button: "p" };
  const component = renderer.create(<ComponentsProvider value={contextComponents}><MyWrappedComponent components={instanceComponents} /></ComponentsProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("prefixed key overwrites main context key", () => {
  const contextComponents = { Button: "div", MyComponent_Button: "p" }; /* eslint-disable-line camelcase */
  const component = renderer.create(<ComponentsProvider value={contextComponents}><MyWrappedComponent /></ComponentsProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("no errors when components context is missing", () => {
  const instanceComponents = { Button: "p" };
  const component = renderer.create(<MyWrappedComponent components={instanceComponents} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("no errors and `components` prop is null when all components are missing", () => {
  const component = renderer.create(<MyWrappedComponent />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("preserves non-React static properties", () => {
  expect(MyWrappedComponent.someStaticProp).toBe("foo");
});

test("wraps display name", () => {
  expect(MyWrappedComponent.displayName).toBe("WithComponents(MyComponent)");
});
