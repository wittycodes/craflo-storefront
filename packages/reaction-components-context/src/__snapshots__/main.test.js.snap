// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components prop key overwrites context key 1`] = `"{\\"Button\\":\\"p\\"}"`;

exports[`merges components from context with components prop 1`] = `"{\\"abc\\":\\"123\\",\\"xyz\\":\\"321\\"}"`;

exports[`no errors and \`components\` prop is null when all components are missing 1`] = `"null"`;

exports[`no errors when components context is missing 1`] = `"{\\"Button\\":\\"p\\"}"`;

exports[`passes components from context in components prop 1`] = `"{\\"foo\\":\\"bar\\"}"`;

exports[`prefixed key overwrites main context key 1`] = `"{\\"Button\\":\\"p\\",\\"MyComponent_Button\\":\\"p\\"}"`;
